import/index_java/src/org/vufind/index/FormatCalculator.java

```diff
@@ -261,26 +287,108 @@ protected String getFormatFromBibLevel(Record record, char bibLevel, char format
+                switch (get008Value(marc008, 33)) {
+                    case 'f':
+                        return "Filmstrip";
+                    case 't':
+                        return "Transparency";
+                    case 'm':
+                        return "MotionPicture";
+                    case 'v': // Videorecording
+                        return "Video";
+                    default: break;
+                }
+                // Check 008/34 Technique
+                // If set, this is a video rather than a slide
+                switch (get008Value(marc008, 34)) {
+                    case 'a': // Animation
+                    case 'c': // Animation and live action
+                    case 'l': // Live action
+                    case 'u': // Unknown
+                    case 'z': // Other
+                        return "Video";
+                    default: break;
+                }
+                // Insufficient info in LDR and 008 to distinguish still from moving images
+                // If there is a 007 for either "Projected Graphic", "Motion Picture", or "Videorecording"
+                // that should contain more information, so return nothing here.
+                // If no such 007 exists, fall back to "ProjectedMedium"
+                if (formatCodes007.contains('g') || formatCodes007.contains('m') || formatCodes007.contains('v')) {
+                    return "";
+                } else {
+                    return "ProjectedMedium";
+                }
             case 'i':
                 return "SoundRecording";
             case 'j':
                 return "MusicRecording";
             case 'k':
-                return "Photo";
+                // Check 008/33 Type of visual material
+                switch (get008Value(marc008, 33)) {
+                    case 'l': // Technical drawing
+                        return "Drawing";
+                    case 'n':
+                        return "Chart";
+                    case 'o':
+                        return "FlashCard";
+                    default: break;
+                }
+                // Insufficient info in LDR and 008 to distinguish image types
+                // If there is a 007 for Nonprojected Graphic, it should have more info, so return nothing here.
+                // If there is no 007 for Nonprojected Graphic, fall back to "Image"
+                return (formatCodes007.contains('k')) ? "" : "Image";
+            // Computer file
+            case 'm':
+                // Check 008/26 Type of computer file
+                switch (get008Value(marc008, 26)) {
+                    case 'a': // Numeric data
+                        return "Dataset";
+                    case 'b': // Computer program
+                        return "Software";
+                    case 'c': // Representational
+                        return "Image";
+                    case 'd': // Document
+                        return "Software";
```

Comment:
@demiankatz 'Text' might be better here, or a new value?

Reply:
I guess the question is, is "Document" a subset of "Text" or vice versa? I wonder if it would make sense to use "Document" everywhere that "Text" is currently used. But ultimately it doesn't matter much since it can be easily remapped... but I do agree that "Software" is probably not as good as "Document" or "Text" here.

Reply:
Right, I think I'll go with "Document" here so as I suppose "document" here could include spreadsheet, powerpoint, etc. People can map that as they wish then, e.g. to "Electronic Document" or to "Text". 

I was thinking of "Text" as fall-back similar to "Image", in the same way as those broad categories are used by DPLA, Europeana, DataCite. Another query I had on this, was if "Text" should take over from things like "Book" in the format_map_level0.properties, again to be more consistent with how "Image" is used there. I didn't want to make that change though, as we don't use these hierarchical facets, so I don't have a good sense of how they work.

Reply:
Sounds reasonable to me -- so I guess we're adding two new formats, "Document" and "Text."

As for whether to use Text in the hierarchical formats, that might be useful, but I think it depends on local preferences. Probably best to leave it as-is for now, but it would be very easy to remap as needed. @EreMaijala might have stronger opinions than me, though, since he set that up originally.

Reply:
set to ComputerDocument and mapped to Document in format_map